File: python-benchmarks/benchmarks/json_benchmark.py
"""
JSON Benchmark
===============
Benchmark for the standard JSON format (text-based).
Uses Python's built-in json module.
"""

import json
from .base_benchmark import BaseBenchmark


class JsonBenchmark(BaseBenchmark):
    """Benchmark for JSON serialization/deserialization."""

    @property
    def format_name(self) -> str:
        return "JSON"

    def serialize(self, data: dict) -> bytes:
        """Serialize dict to JSON bytes (UTF-8)."""
        return json.dumps(data, ensure_ascii=False).encode("utf-8")

    def deserialize(self, payload: bytes) -> dict:
        """Deserialize JSON bytes to dict."""
        return json.loads(payload.decode("utf-8"))

File: go-benchmarks/benchmarks/json_benchmark.go
package benchmarks

import (
	"encoding/json"
)

// JSONBenchmark benchmarks the standard JSON format.
type JSONBenchmark struct{}

func (b *JSONBenchmark) FormatName() string {
	return "JSON"
}

func (b *JSONBenchmark) Serialize(data map[string]interface{}) ([]byte, error) {
	return json.Marshal(data)
}

func (b *JSONBenchmark) Deserialize(payload []byte) (map[string]interface{}, error) {
	var result map[string]interface{}
	err := json.Unmarshal(payload, &result)
	return result, err
}
